src/control/server/server_utils.go

```diff
@@ -540,6 +540,29 @@ func checkEngineTmpfsMem(srv *server, ei *EngineInstance, mi *common.MemInfo) er
 	memRamdisk := uint64(sc.Scm.RamdiskSize) * humanize.GiByte
 	memAvail := uint64(mi.MemAvailableKiB) * humanize.KiByte
 
+	// In the event that tmpfs was already mounted, we need to verify that it
+	// is the correct size and that the memory usage still makes sense.
+	if isMounted, err := ei.storage.ScmIsMounted(); err == nil && isMounted {
+		usage, err := ei.storage.GetScmUsage()
+		if err != nil {
+			return errors.Wrap(err, "unable to check tmpfs usage")
+		}
+		if usage.TotalBytes != memRamdisk {
+			// Check that the existing ramdisk is at least 90% of the
+			// calculated optimal ramdisk size and not larger.
+			if usage.TotalBytes < memRamdisk*90/100 {
```

Comment:
This first check should be replaced by `if err := checkMemForRamdisk(log, memRamdisk, usage.TotalBytes); err != nil { return err } else if { ...` to avoid duplication and "magic" numbers.

Reply:
Resolved with removal of minimum size check.